include/flexflow/model.h

```diff
@@ -639,35 +641,7 @@ class FFModel {
   }
 
   template <typename T>
-  PCG::Node get_or_create_node(const std::vector<ParallelTensor>& inputs, typename T::Params const& params) {
-    using Params = typename T::Params;
-
-    ParallelTensorShapes input_shapes;
-    for (const auto& input : inputs) {
-      input_shapes.push_back(input->get_shape());
-    }
-
-    T* op = nullptr;
-
-    std::pair<ParallelTensorShapes, Params> key{input_shapes, params};
-    auto &cache = this->get_cache_multi_inputs<T>();
-    const auto &it = cache.find(key);
-    if (it != cache.end()) {
-      op = it->second;
-    } else {
-      op = new T(*this, params, inputs, NULL);
-      cache[key] = op;
-    }
-
-    assert(op->get_params() == params);
-    return this->new_node(op);
-  }
-
-  template <typename T>
-  std::unordered_map<std::pair<ParallelTensorShape, typename T::Params>, T*> &get_cache();
-
-  template <typename T>
-  std::unordered_map<std::pair<ParallelTensorShapes, typename T::Params>, T*> &get_cache_multi_inputs();
+  std::unordered_map<std::pair<typename T::InputShapeType, typename T::Params>, T*> &get_cache();
```

Comment:
```suggestion
  std::unordered_map<std::pair<typename T::InputShapeType, typename T::Params>, T*> &get_cache() = delete;
```
Apparently this causes use of a nonexistent specialization to throw a compiler error rather than a linker error.